app/controllers/Tournament.scala

```diff
@@ -302,4 +319,19 @@ object Tournament extends LilaController {
     },
     expireAfter = _.ExpireAfterWrite(15.seconds)
   )
+
+  private val teamApi = lila.app.Env.team.api
+  private def teamIdsIBelongTo(me: lila.user.User): Fu[List[String]] =
+    teamApi.mine(me) map { teams =>
+      teams.map(_.id)
+    }
+  private def teamsIBelongTo(me: lila.user.User): Fu[List[(String, String)]] =
+    teamApi.mine(me) map { teams =>
+      teams.map(t => t._id -> t.name)
+    }
+  private def teamIdsIBelongToO(me: Option[lila.user.User]): Fu[Option[List[String]]] =
```

Comment:
taking an optional argument is usually a bad idea. Instead, if `me` is an `Option[lila.user.User]`, you can use the `teamsIBelongTo` function as such: `me ?? teamsIBelongTo`

Reply:
I'm having trouble figuring out how to do that; it's easy to create a Option[Fu[...]] but Fu[Option[...]] seems to be more complicated - which is why I originally created teamIdsIBelongToO.

Reply:
I did figure out a one-liner with `fold` but I don't know if it's good to repeat it 3 times (that's how often I use teamIdsIBelongToO) instead of having it as one function.

Reply:
Nevermind, I was able to translate the `fold` to `??` as you initially suggested.